lib/stats-collector.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+/**
+ * Initializes test statistics collector object.
+ *
+ * Provides stats such as test duration,
+ * number of tests passed / failed etc.
+ *
+ * @public
+ * @memberof Mocha
+ * @param {Runner} runner
+ * @return {StatsCollector}
+ *
+ * Reporter statistics.
+ *
+ * @typedef {Object} StatsCollector
+ * @property {number} suites - integer count of suites run.
+ * @property {number} tests - integer count of tests run.
+ * @property {number} passes - integer count of passing tests.
+ * @property {number} pending - integer count of pending tests.
+ * @property {number} failures - integer count of failed tests.
+ * @property {Date} start - time when testing began.
+ * @property {Date} end - time when testing concluded.
+ * @property {number} duration - number of msecs that testing took.
+ */
+function initStatsCollector(runner) {
```

Comment:
`function createStatsCollector(runner) {`

Sorry for the naming waffling here...